### ISSUE TITLE
Nice to have: add proxy support for announcers

### ISSUE THREAD

**Issue by zerg518** [transmission/transmission#344]
For different reasons announcer servers of some trackers are blocked by ISP. Torrent traffic is not blocked.
So it will be good if transmission will use proxy just for communication with announcer servers, not for torrent traffic.

**Comment by jm134** [comment]
(y)

**Comment by mikedld** [comment]
Proxy support has a long history in Transmission...

Current, you could pass `http_proxy` environment variable (or any other variable outlined in [curl documentation](https://curl.haxx.se/docs/manpage.html) under ENVIRONMENT or elsewhere) to any Transmission client or the daemon to make it use proxy server for scrapes and announces. In fact, `http_proxy` is even mentioned in some of the respective man pages.

That's where it ends for now. There are no plans to support proxy more extensively, but if you want that to happen --- feel free to discuss here and make a pull request (which may or may not be accepted depending on its quality and demand).
